### ISSUE TITLE
Trying to be more "Agile"?

### ISSUE THREAD

**Issue by G0maa** [G0maa/my-school-server#17]
These are *some* notes taken from reading Martin Fowler's "Is Design Dead?" article:
https://www.martinfowler.com/articles/designDead.html
(To add more resources/articles?)

```
NO DUPLICATION.
YAGNI.
DRY.
```

- I would say **current goal** is to generally do the "enabling practices" of Extreme Programming, which are (as mentioned in the article):
  - (Automated) Testing
  - Continuous Integration
  - Refactoring
  - Simple Design
- Many of said things are _very_ subjective.
- **tl;dr To-Do & Keep an eye on these four things.**
- About Simple Design:
  - Runs all tests,
  - No duplication,
  - ?Reveals all intention?
  - Fewest number of classes or methods,

**Comment by G0maa** [comment]
How I notice possible `WET` code? => ez pz just comment on it 😬

**Comment by G0maa** [comment]
CRUD logic is very duplicate(able).

**Comment by G0maa** [comment]
* Adding a "services" layer, would indeed ?greatly? summarize most of the code I wrote?
* Looks like the only thing I'm doing is `refactoring` without `simple design`.... 🤦🏻‍♂️🤦🏻‍♂️

**Comment by G0maa** [comment]
- Thinking of refactoring to OOP, instead of the current imperative style.
